File: CS/DetailViewButton.Module/Updater.cs
using System;

using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;

namespace DetailViewButton.Module {
    public class Updater : ModuleUpdater {
        public Updater(Session session, Version currentDBVersion) : base(session, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();

            Person testPerson = Session.FindObject<Person>(CriteriaOperator.Parse("FirstName == 'Sam'"));
            if(testPerson == null) {
                testPerson = new Person(Session);
                testPerson.FirstName = "Sam";
                testPerson.Save();
            }
        }
    }
}

File: VB/DetailViewButton.Module/Updater.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.BaseImpl

Namespace DetailViewButton.Module
	Public Class Updater
		Inherits ModuleUpdater
		Public Sub New(ByVal session As Session, ByVal currentDBVersion As Version)
			MyBase.New(session, currentDBVersion)
		End Sub
		Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
			MyBase.UpdateDatabaseAfterUpdateSchema()

			Dim testPerson As Person = Session.FindObject(Of Person)(CriteriaOperator.Parse("FirstName == 'Sam'"))
			If testPerson Is Nothing Then
				testPerson = New Person(Session)
				testPerson.FirstName = "Sam"
				testPerson.Save()
			End If
		End Sub
	End Class
End Namespace
